### PR TITLE
Commerce 8228

### PR DESCRIPTION
**Pull request by MahmoudAzzam**

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java [hunk 1]
```diff
@@ -175,13 +175,15 @@ public CPInstance addCPInstance(
 		cpInstance.setExpirationDate(expirationDate);
 		cpInstance.setOverrideSubscriptionInfo(overrideSubscriptionInfo);
 		cpInstance.setSubscriptionEnabled(subscriptionEnabled);
-		cpInstance.setSubscriptionLength(subscriptionLength);
+		cpInstance.setSubscriptionLength(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:178]: I don't like this, can we move this to Action Command level?

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java [hunk 2]
```diff
@@ -175,13 +175,15 @@ public CPInstance addCPInstance(
...
+			(subscriptionLength == 0) ? 1 : subscriptionLength);
 		cpInstance.setSubscriptionType(subscriptionType);
 		cpInstance.setSubscriptionTypeSettingsProperties(
 			subscriptionTypeSettingsUnicodeProperties);
 		cpInstance.setMaxSubscriptionCycles(maxSubscriptionCycles);
 		cpInstance.setDeliverySubscriptionEnabled(deliverySubscriptionEnabled);
-		cpInstance.setDeliverySubscriptionLength(deliverySubscriptionLength);
+		cpInstance.setDeliverySubscriptionLength(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:185]: same here

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
Ehi @alessiorendina can you have a look at this too?

**Comment by alessiorendina** [comment]
@andrea-ale-sbarra I agree with your suggestions. The other commit looks good to me.

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:relevant

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
I tested this PR manually and buy running  CPCommerceProductSKUs#AddASKUWithSubscriptions. The issue is fixed. After enabling the "Override Subscription Settings" toggle and saving, a green successful alert appears, and if you refresh the page the settings are saved.

**Comment by Kvale1733** [comment]
ci:forward

**Pull request closed without merging**